src/Networking/NexusMods.Networking.HttpDownloader/AdvancedHttpDownloader.cs

```diff
@@ -414,6 +411,7 @@ private async Task ReadStreamToEnd(Stream stream, ChunkState chunk, ChannelWrite
             {
                 byte[] buffer = _bufferPool.Rent(ReadBlockSize);
                 lastRead = await stream.ReadAsync(buffer, 0, ReadBlockSize, cancel);
+                job.ReportNoWait(Size.From(lastRead));
```

Comment:
I think this needs to be the wait version so we respect bandwidth limiting, but I want to do some testing with that and make sure it doesn't mess up the rest of the code.